gitreceive/receiver/flynn-receive.go

```diff
@@ -189,23 +199,28 @@ Options:
 			Processes: map[string]int{"web": 1},
 		}
 
-		watcher, err := client.WatchJobEvents(app.ID, release.ID)
-		if err != nil {
-			log.Fatalln("Error streaming job events", err)
-			return
-		}
-		defer watcher.Close()
-
 		if err := client.PutFormation(formation); err != nil {
 			log.Fatalln("Error putting formation:", err)
 		}
-		fmt.Println("=====> Waiting for web job to start...")
 
-		err = watcher.WaitFor(ct.JobEvents{"web": {"up": 1}}, scaleTimeout, func(e *ct.Job) error {
-			switch e.State {
-			case "up":
-				fmt.Println("=====> Default web formation scaled to 1")
-			case "down", "crashed":
+		expected = ct.JobEvents{"web": {"up": 1}}
+	} else {
+		formation, err := client.GetFormation(app.ID, release.ID)
+		if err != nil {
+			log.Fatalln("Error getting formation:", err)
+		}
+		expected = make(ct.JobEvents, len(formation.Processes))
+		for k, v := range formation.Processes {
+			if k == "web" {
```

Comment:
I think this will work:

```go
var expected ct.JobEvents
if n, ok := formation.Processes["web"]; ok && n > 0 {
    expected = ct.JobEvents{"web": map[string]int{"up": v}}
}
```